lib/ros_message.h

```diff
@@ -11,16 +11,29 @@
 namespace Embag {
 class RosMessage {
  public:
-  std::string topic;
-  RosValue::ros_time_t timestamp;
-  std::string md5;
+  const std::string topic;
+  const RosValue::ros_time_t timestamp;
+  const std::string md5;
   const std::shared_ptr<std::vector<char>> raw_buffer;
   const size_t raw_buffer_offset;
-  uint32_t raw_data_len = 0;
-
-  RosMessage(const std::shared_ptr<std::vector<char>> raw_buffer, const size_t offset)
-    : raw_buffer(raw_buffer)
+  const uint32_t raw_data_len;
+
+  RosMessage(
+    const std::string& topic,
+    const RosValue::ros_time_t& timestamp,
+    const std::string& md5,
```

Comment:
I'm a little worried that adding these args at the front could cause backwards compatibility issues. Is this a class that is only used internal to embag?

Reply:
Theoretically, it would be kind of weird to create the class externally since the `msg_def` field is private and theres no actual way to get the data without it. I can remove this change though as its not necessary, though it would just be good cleanup

Reply:
I just removed it!